src/_premake_init.lua

```diff
@@ -1512,6 +1512,17 @@
 		tokens = true,
 	}
 
+	api.register {
+		name = "enableUnityBuild",
+		scope = { "config", "rule" },
```

Comment:
`"rule"` seems suspicious...

Reply:
Initially when I added this, "config" didn't work, so I added "rule" and that seemed to work. didn't find enough documentation around what it really means but also didn't look that hard.